File: princeton/py/8threesumbrute.py
# CLI usage: 
# python 8threesumbrute.py 4 -4 2 6 0 -2 8 10 -6 ( WILL NOT USE CLI)
#
# File usage:
#4 -4 2 6 0 -2 8 10 -6

#
# Brute force implementation of Three-sum
# Read an array of ints
# Identify all triples that sum to zero
#

import sys, importlib.util

INPUT_FILE_PATH = '/home/ubuntu/Dev/algs-ds/input/8threesum1.txt'

specFile = importlib.util.spec_from_file_location('file', '/home/ubuntu/Dev/algs-ds/tools/file.py')
file = importlib.util.module_from_spec(specFile)
specFile.loader.exec_module(file)

specCli = importlib.util.spec_from_file_location('cli', '/home/ubuntu/Dev/algs-ds/tools/cli.py')
cli = importlib.util.module_from_spec(specCli)
specCli.loader.exec_module(cli)

class threesumbrute:
    
    def __init__(self):
        self.counter = 0

    def threesum(self, listA):
        for i in range(len(listA)):
            for j in range(i + 1, len(listA)):
                for k in range(j + 1, len(listA)):
                    if (listA[i] + listA[j] + listA[k] == 0):
                        print("3-SUM: {} {} {}".format(listA[i], listA[j], listA[k]))
                        self.counter += 1
        print('count: {}'.format(self.counter))


def runClient(inputs):
    listA = inputs[0]

    tsb = threesumbrute()
    tsb.threesum(listA)

##
## CLIENT
##

if len(sys.argv) > 1:
    runClient(cli.readArgTo2DArray(sys.argv[1]))
else:
    runClient(file.readLinesTo2DArray(INPUT_FILE_PATH))

File: princeton/js/8threesumbrute.js
// CLI usage: 
// node 8threesumbrute.js 4 -4 2 6 0 -2 8 10 -6 ( WILL NOT USE CLI)
//
// File usage:
//4 -4 2 6 0 -2 8 10 -6

//
// Brute force implementation of Three-sum
// Read an array of ints
// Identify all triples that sum to zero
//

var file = require('/home/ubuntu/Dev/algs-ds/tools/file.js');
var cli = require('/home/ubuntu/Dev/algs-ds/tools/cli.js');

const INPUT_FILE_PATH = '/home/ubuntu/Dev/algs-ds/input/8threesum3.txt'

class threesumbrute { 
    constructor(list) {
        this.list = list;
    }

    threesum(list) {
        let n = list.length;
        let count = 0;
        for (let i = 0; i < n; i++) {
            for (let j = i + 1; j < n; j++) {
                for (let k = j + 1; k < n; k++) {
                    if (list[i] + list[j] + list[k] === 0) {
                        console.log('3-SUM: ' + list[i] + ' ' + list[j] + ' ' + list[k]);
                        count += 1;
                    }
                }
            }
        }
        console.log('count: ' + count);
    }
}

function runClient (inputs) {
    let list = inputs[0];

    let tsb = new threesumbrute();
    tsb.threesum(list);
}

//
// CLIENT
//

if (process.argv.length > 2) {
    cli.readArgTo2DArray(process.argv[2], function (cliInput) {
         runClient(cliInput);
    });
 
 } else {
     file.readLinesTo2DArray(INPUT_FILE_PATH, function (fileInput) {
         runClient(fileInput);
     });
 }